paper-dialog-behavior.html

```diff
@@ -205,32 +205,36 @@
     },
 
     _onFocus: function(event) {
-      if (this.modal) {
```

Comment:
yes i agree and I've put back the check for modal.
There are still other issues related to the change of modal while the dialog is opened though, e.g. backdrop is not removed, listeners are not toggled. This stuff should be handled in a separate PR